specifications/cloud_assembly.md

```diff
@@ -0,0 +1,398 @@
+`stackPolicy`|`string`            |        |The assembly-relative path to the [Stack Policy][CFN Stack Policy].
+
+##### Output Attributes
+Attribute |Type                |Description
+----------|--------------------|-----------
+`outputs` |`Map<string,string>`|Data returned by [*CloudFormation* Outputs][CFN Outputs] of the stack.
+`stackArn`|`string`            |The ARN of the [stack][CFN Stack].
+
+##### Example
+```json
+{
+  "type": "npm://@aws-cdk/aws-cloudformation.StackDrop",
+  "environment": "aws://000000000000/bermuda-triangle-1",
+  "properties": {
+    "template": "my-stack/template.yml",
+    "parameters": {
+      "bucketName": "${helperStack.bucketName}",
+      "objectKey": "${helperStack.objectKey}"
+    },
+    "stackPolicy": "my-stack/policy.json"
+  }
+}
+```
+
+[CFN Stack]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacks.html
+[CFN Stack Policy]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/protect-stack-resources.html
+[CFN Outputs]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html
+
+#### `@aws-cdk/assets.FileDrop`
+A file that needs to be uploaded to an *S3* bucket.
+
+##### Properties
+Property    |Type    |Required|Description
+------------|--------|:------:|-----------
+`file`      |`string`|Required|The assembly-relative path to the file that will be uploaded.
+`bucketName`|`string`|Required|The name of the bucket where this file will be uploaded.
+`objectKey` |`string`|Required|The key at which to place the object in the bucket.
+
+##### Output Attributes
+Attribute   |Type    |Description
+------------|--------|-----------
+`bucketName`|`string`|The name of the bucket where the file was uploaded.
+`objectKey` |`string`|The key at which the file was uploaded in the bucket.
+
+##### Example
+```json
+{
+  "type": "npm://@aws-cdk/assets.FileDrop",
+  "environment": "aws://000000000000/bermuda-triangle-1",
+  "properties": {
+    "file": "assets/file.bin",
+    "bucket": "${helperStack.bucketName}",
+    "objectKey": "assets/da39a3ee5e6b4b0d3255bfef95601890afd80709/nifty-asset.png"
+  }
+}
+```
+
+#### `@aws-cdk/aws-ecr.DockerImageDrop`
+A Docker image to be published to an *ECR* registry.
+
+##### Properties
+Property    |Type    |Required|Description
+------------|--------|:------:|-----------
+`savedImage`|`string`|Required|The assembly-relative path to the tar archive obtained from `docker image save`.
+`imageName` |`string`|Required|The name of the image (e.g: `000000000000.dkr.ecr.bermuda-triangle-1.amazon.com/name`).
+`tagName`   |`string`|        |The name of the tag to use when pushing the image (default: `latest`).
+
+##### Output Attributes
+Attribute     |Type    |Description
+--------------|--------|-----------
+`exactImageId`|`string`|An absolute reference to the published image version (`imageName@DIGEST`).
+`imageName`   |`string`|The full tagged image name (`imageName:tagName`).
```

Comment:
Make this name different from input `imageName` so that it's clear it's not the same schema. `fullImageName`? `qualifiedImageName`?

Reply:
Would `imageId` make sense here? I guess it collides unfortunately with docker terminology...